### DISCUSSION TITLE
skyfield.api.Angle arithmetic ?

### DISCUSSION THREAD

**Discussion by captcha1** [skyfielders/python-skyfield#719] [Q&A]
Shouldn't skyfield.api.Angle have arithmetic methods (addition, subtraction) ...?

Eg, I'm computing altitude w/ altaz, and want to subtract semi-diameter.  Is there an easy way to do this ?

**Comment by brandon-rhodes** [comment]
We are not currently planning to add math operations to `Angle`, `Distance`, or `Velocity`, because full math support involves dozens of methods having to be added to every one of those classes. Instead, we encourage programmers to use specific units like `.degrees`, which return Python floating point objects (or NumPy vectors), both of which already support the full range of Python math operations.

Can you do something like `alt.degrees - semi_diameter_degrees`?

**Comment by captcha1** [comment]
Mmm.  That works.

Is there an easy way to format an angle with "dd mm.m" (common in celestial navigation) ?

I'm no python wizard, but couldn't you define arithmetic in "class Unit" since Angle, etc inherits from Unit ...?

**Reply by brandon-rhodes** [reply]
> Is there an easy way to format an angle with "dd mm.m" (common in celestial navigation) ?

Yes! Here's the class documentation:

https://rhodesmill.org/skyfield/api-units.html

(Let me know if the descriptions are too obscure and I can try expanding them.)

> I'm no python wizard, but couldn't you define arithmetic in "class Unit" since Angle, etc inherits from Unit ...?

Yes, if a level of indirection was introduced: the arithmetic would have to access a different attribute (`au` vs `au_per_d` versus `radians`) for every subclass of `Unit`. All of the resulting complexity would not only laden Skyfield with extra code, but would make something as basic as arithmetic even slower than with per-class dunder methods. So I'm still at the moment convinced that it's better for folks to do their floating-point and NumPy vector operations manually, rather than hiding them behind Skyfield operators.

But, yes — if the code were going to be written, adding it once to `Unit` would be far cleaner than adding it *n* times, once to each subclass! Good point.

**Reply by captcha1** [reply]
I saw the docs.  I see "dd mm ss.s" format but I don't see "dd mm.m" ...

**Reply by brandon-rhodes** [reply]
Oh — I didn't read your question carefully enough, did I? No: there isn't any built-in way to stop at decimal minutes. The next time I'm in that code, I'll think about whether there's an easy way to add support for that use case.
